packages/viewer/src/components/scene-tree/scene-tree.tsx

```diff
@@ -134,25 +209,206 @@ export class SceneTree {
 
-  public componentWillLoad(): void {
+  /**
+   * Schedules a render of the rows in the scene tree. Useful if any custom
+   * data in your scene tree has changed, and you want to update the row's
+   * contents.
+   *
+   * **Note:** This is an asynchronous operation. The update may happen on the
+   * next frame.
+   */
+  @Method()
+  public async invalidateRows(): Promise<void> {
+    forceUpdate(this.el);
+  }
+
+  @Method()
+  public async scrollToIndex(index: number): Promise<void> {
+    // TODO(dan): Add alignment to top, center, or bottom.
+    const i = Math.max(0, Math.min(index, this.totalRows));
+
+    if (this.computedRowHeight != null) {
+      const top = i * this.computedRowHeight;
+      this.el.scrollTo({ top, behavior: 'smooth' });
+    }
+  }
+
+  /**
+   * Performs an API call to expand all nodes in the tree.
+   */
+  @Method()
+  public async expandAll(): Promise<void> {
+    await this.controller?.expandAll();
+  }
+
+  /**
+   * Performs an API call to collapse all nodes in the tree.
+   */
+  @Method()
+  public async collapseAll(): Promise<void> {
+    await this.controller?.collapseAll();
+  }
+
+  /**
+   * Performs an API call that will expand the node associated to the specified
+   * row or row index.
+   *
+   * @param rowOrIndex A row or row index to expand.
+   */
+  @Method()
+  public async expandItem(rowOrIndex: number | Row): Promise<void> {
+    await this.performRowOperation(rowOrIndex, async ({ row }) => {
+      if (!row.expanded) {
+        await this.controller?.expandNode(row.id);
+      }
+    });
+  }
+
+  /**
+   * Performs an API call that will collapse the node associated to the
+   * specified row or row index.
+   *
+   * @param rowOrIndex A row or row index to collapse.
+   */
+  @Method()
+  public async collapseItem(rowOrIndex: number | Row): Promise<void> {
+    await this.performRowOperation(rowOrIndex, async ({ row }) => {
+      if (row.expanded) {
+        await this.controller?.collapseNode(row.id);
+      }
+    });
+  }
+
+  /**
+   * Performs an API call that will either expand or collapse the node
+   * associated to the given row or row index.
+   *
+   * @param rowOrIndex The row or row index to collapse or expand.
+   */
+  @Method()
+  public async toggleExpandItem(rowOrIndex: number | Row): Promise<void> {
+    await this.performRowOperation(rowOrIndex, async ({ row }) => {
+      if (row.expanded) {
+        await this.collapseItem(row);
+      } else {
+        await this.expandItem(row);
+      }
+    });
+  }
+
+  /**
+   * Performs an API call that will either hide or show the item associated to
+   * the given row or row index.
+   *
+   * @param rowOrIndex The row or row index to toggle visibility.
+   */
+  @Method()
+  public async toggleItemVisibility(rowOrIndex: number | Row): Promise<void> {
+    await this.performRowOperation(rowOrIndex, async ({ viewer, row }) => {
+      if (row.visible) {
+        await hideItem(viewer, row.id);
```

Comment:
Any reason you couldn't just return hideItem instead of awaiting it here? Not a big deal, just curious if there was a reason.

Reply:
You can, it's effectively the same. Maybe one benefit to this, is if the return signature for `hideItem` changes, this code doesn't need to change.